src/acl.c

```diff
@@ -2002,18 +2108,33 @@ sds ACLLoadFromFile(const char *filename) {
          * be cleanly applied to the user. If any option fails
          * to apply, the other values won't be applied since
          * all the pending changes will get dropped. */
-        int j;
-        for (j = 2; j < argc; j++) {
-            argv[j] = sdstrim(argv[j],"\t\r\n");
-            if (ACLSetUser(u,argv[j],sdslen(argv[j])) != C_OK) {
+        int merged_argc;
+        sds *acl_args = ACLMergeSelectorArguments(argv + 2, argc - 2, &merged_argc, NULL);
+        if (!acl_args) {
+            errors = sdscatprintf(errors,
+                    "%s:%d: Unmatched parenthesis in selector definition. ",
+                    server.acl_filename, linenum);
+        }
+
+        int j, error = 0;
+        for (j = 0; j < merged_argc; j++) {
+            acl_args[j] = sdstrim(acl_args[j],"\t\r\n");
+            serverLog(LL_WARNING, "%s", acl_args[j]);
+            if (ACLSetUser(u,acl_args[j],sdslen(acl_args[j])) != C_OK) {
                 const char *errmsg = ACLSetUserStringError();
                 errors = sdscatprintf(errors,
                          "%s:%d: %s. ",
                          server.acl_filename, linenum, errmsg);
-                continue;
+                error = 1;
+                break;
```

Comment:
Two things:
1) I don't know why I added error variable, that looks like a bug.
2) The previous version printed out a string that is the filename + linenumber + errmsg for each error, which can be confusing since it's not helpful. Let me pull this change out into a separate PR, since for some definition it is a backwards incompatible change.